stockflux-securities-master/src/components/SecuritiesTable.js

```diff
@@ -1,24 +1,28 @@
-import React, { useEffect, useState, useCallback } from "react";
+import React, { useEffect, useState, useCallback, useReducer } from "react";
 import Components from "stockflux-components";
 import { Link } from "react-router-dom";
 import "./SecuritiesTable.css";
 import * as service from "../services/SecuritiesService";
 import ValidationError from "../services/ValidationError";
 import Alert from "./Alert";
 import { TableState } from "../enums";
+import ToolTip from "./ToolTip";
 import Button from "./Button";
+import { FaPen, FaTrashAlt, FaEye, FaEyeSlash } from "react-icons/fa";
+import { MdClose, MdCheck } from "react-icons/md";
+import {securitiesTableReducer, initialTableState} from "../reducers/securitiesTableReducer";
 
 const SecuritiesTable = ({ location }) => {
+  
   const [securitiesData, setSecuritiesData] = useState([]);
-  const [messages, setMessages] = useState([]);
-  const [state, setState] = useState(TableState.LOADING);
+  const [state, dispatch] = useReducer(securitiesTableReducer, initialTableState);
 
   const handleError = err => {
-    setState(TableState.ERROR);
+    dispatch({type: "error"});
```

Comment:
Is it necessary t have the first dispatch `dispatch({type: "error"});` when error will be dispatched again from if/else below? Maybe this is redundant?